macros/edr/data_monitoring/monitors_query/dimension_monitoring_query.sql

```diff
@@ -0,0 +1,263 @@
+{% macro dimension_monitoring_query(monitored_table_relation, dimensions, where_expression, timestamp_column, is_timestamp, min_bucket_start) %}
+    {% set metric_name = 'dimension' %}
+    {%- set max_bucket_end = "'"~ elementary.get_run_started_at().strftime("%Y-%m-%d 00:00:00")~"'" %}
+    {%- set min_bucket_end = "'"~ (modules.datetime.datetime.strptime(min_bucket_start, "'%Y-%m-%d %H:%M:%S'") +  modules.datetime.timedelta(1)).strftime("%Y-%m-%d 00:00:00")~"'" %}
+    {%- set max_bucket_start = "'"~ (elementary.get_run_started_at() - modules.datetime.timedelta(1)).strftime("%Y-%m-%d 00:00:00")~"'" %}
+    {% set full_table_name_str = "'"~ elementary.relation_to_full_name(monitored_table_relation) ~"'" %}
+    {% set dimensions_saveable_format = elementary.join_list(dimensions, '; ') %}
+    {% set concat_dimensions_sql_expression = elementary.list_concat_with_separator(dimensions, '; ') %}
+    
+    {% if is_timestamp %}
+        with filtered_monitored_table as (
+            select *,
+                   {{ concat_dimensions_sql_expression }} as dimension_value,
+                   {{ elementary.date_trunc('day', timestamp_column) }} as start_bucket_in_data
+            from {{ monitored_table_relation }}
+            where
+                {{ elementary.cast_as_timestamp(timestamp_column) }} >= {{ elementary.cast_as_timestamp(min_bucket_start) }}
+                and {{ elementary.cast_as_timestamp(timestamp_column) }} <= {{ elementary.cast_as_timestamp(max_bucket_end) }}
+            {% if where_expression %}
+                and {{ where_expression }}
+            {% endif %}
+        ),
+
+        {# Outdated dimension values are dimensions with all metrics of 0 in the range of the test time #}
+        filtered_outdated_dimension_values as (
+            select distinct 
+                dimension_value,
+                sum(metric_value)
+            from {{ ref('data_monitoring_metrics') }}
+            where full_table_name = {{ full_table_name_str }}
+                and metric_name = {{ "'" ~ metric_name ~ "'" }}
+                and dimension = {{ "'" ~ dimensions_saveable_format ~ "'" }}
+                and {{ elementary.cast_as_timestamp('bucket_end') }} >= {{ elementary.cast_as_timestamp(min_bucket_start) }}
+            group by 1
+            having sum(metric_value) > 0
+        ),
+
+        all_dimension_values as (
```

Comment:
I would change to 'dimension_values_union', as it's not all (we filtered outdated out).